Civi/Core/SettingsMetadata.php

```diff
@@ -145,20 +151,41 @@ protected static function loadSettingsMetadata($metaDataFolder) {
    *   Metadata to filter.
    */
   protected static function _filterSettingsSpecification($filters, &$settingSpec) {
-    if (empty($filters)) {
-      return;
-    }
-    elseif (array_keys($filters) == ['name']) {
-      $settingSpec = [$filters['name'] => \CRM_Utils_Array::value($filters['name'], $settingSpec, '')];
-      return;
+    if (!empty($filters['name'])) {
+      $settingSpec = array_intersect_key($settingSpec, array_flip((array) $filters['name']));
+      // FIXME: This is a workaround for settingsBag::setDb() called by unit tests with settings names that don't exist
+      $settingSpec += array_fill_keys((array) $filters['name'], []);
```

Comment:
@eileenmcnaughton I had to add this line to mimic the old behavior.
Settings code is kind of a mess here. Settings::set() calls a validate function that doesn't actually work, and unit tests are counting on that so they can test make-believe settings :/

Reply:
FYI this is why the validate function doesn't work: If you pass in `['name' => 'foo']` as the `$filters` to this function then it will spit back out an array with the key foo *even if there is no such setting!*. IMO that's wrong, but out of scope for this PR so I left the errant behavior as-is.